### ISSUE TITLE
.NET 6 target isn't currently tested

### ISSUE THREAD

**Issue by Neme12** [dotnet/roslyn#69360]
I noticed that projects like Microsoft.CodeAnalysis currently target netstandard2.0, net6.0 and net7.0, while the corresponding unit tests target net472, net7.0 and net8.0, which means that the net6.0 target isn't tested at all, which could be a problem if there are any #if directives based on that.

**Comment by Neme12** [comment]
@jaredpar Is this expected? This seems really bad to me to have a target that is completely untested.

**Comment by jaredpar** [comment]
That is correct. The full set of TFM that our code builds for and executes on is ... 

- `net472`: Visual Studio and desktop MSBuild scenarios
- `net6.0`: Visual Studio private runtime 
- `net7.0`: .NET SDK 7.0 builds, VS Code, source build
- `net8.0`: .NET SDK 8.0 builds, source build

Our code also builds for Windows, Linux and Mac. 

For a long time we held the line at a much smaller set of TFM however the product lines we participate in have forced us to expand. The resulting test matrix for all these different TFM and platforms is _quite_ large. 

When our TFM list expanded to the current set we discussed what it would take to effectively put the full combination into our CI loop _and_ what value add we would get from doing this. The result is it would significantly complicate our CI setup and we felt the value add for doing so was very small. 

The reason the value add is small is historically we've seen very few issues when executing on older runtimes. Our issues instead happen when migrating to new runtimes where subtle behavior changes we depend on change. After getting past those our execution on a runtime is fairly stable. Basically if our tests run fine on `net7.0` it is a very strong indicator they will also run on `net6.0` just fine. The number of times this has not held is vanishingly small. 

As such our decision was to go with the setup we have now. If we found that we were shipping regressions, having issues, etc ... we could revisit and invest in putting outer loop tests for other TFMs.
